pydeseq2/dds.py

```diff
@@ -477,7 +477,7 @@ def fit_size_factors(
             warnings.warn(
                 "Every gene contains at least one zero, "
                 "cannot compute log geometric means. Switching to iterative mode.",
-                RuntimeWarning,
+                UserWarning,
```

Comment:
Out of curiosity, what is the rationale of selecting UserWarning ?

Reply:
The bad reason: it was easier to catch it simultaneously with a switch to "mean" trend curve fitting in the tests. The better reason is that I felt it was more suited (`RuntimeWarning` is supposed to cover [dubious runtime behavior](https://docs.python.org/3/library/exceptions.html#RuntimeWarning) which is not really what's going on here).